target-chains/cosmwasm/tools/src/deploy-pyth-bridge.ts

```diff
@@ -0,0 +1,133 @@
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+import { Deployer, DeployerFactory } from "./deployer";
+import { NETWORKS_OPTIONS } from "./network";
+import { PythConfig } from "./pyth_config";
+
+const argv = yargs(hideBin(process.argv))
+  .option("network", {
+    description: "Which network to deploy to",
+    choices: NETWORKS_OPTIONS,
+    required: true,
+  })
+  .option("artifact", {
+    description: "Path to Pyth artifact",
+    type: "string",
+    required: false,
+  })
+  .option("mnemonic", {
+    description: "Mnemonic (private key)",
+    type: "string",
+    required: true,
+  })
+  .option("instantiate", {
+    description: "Instantiate contract if set (default: disabled)",
+    type: "boolean",
+    default: false,
+    required: false,
+  })
+  .option("migrate", {
+    description: "Migrate an existing contract if set (default: disabled)",
+    type: "boolean",
+    default: false,
+    required: false,
+  })
+  .option("contract", {
+    description: "Contract address, used only for migration",
+    type: "string",
+    required: false,
+    default: "",
+  })
+  .option("code-id", {
+    description:
+      "Code Id, if provided this will be used for migrate/instantiate and no code will be uploaded",
+    type: "number",
+    requred: false,
+  })
+  .help()
+  .alias("help", "h")
+  .parseSync();
+
+const {
+  artifact,
+  network,
+  mnemonic,
+  codeId: inputCodeId,
+  instantiate,
+  migrate,
+  contract,
+} = argv;
+const pythConfig = PythConfig[network];
+const deployer: Deployer = DeployerFactory.create(network, mnemonic);
+
+// checks
+if (inputCodeId === undefined && artifact === undefined)
+  RaiseCLError("Please provide either artifact or code id");
+
+async function deployArtifacts(artifact: string): Promise<number> {
```

Comment:
these three helper functions seem unnecessary it's more code to have them than to call the function on `deployer` directly in the `run` function below